src/ResourceManager/Sql/Commands.Sql/Elastic Pools/Services/AzureSqlElasticPoolAdapter.cs

```diff
@@ -342,29 +349,62 @@ public string GetServerLocation(string resourceGroupName, string serverName)
         /// <returns>The converted model</returns>
         private AzureSqlElasticPoolModel CreateElasticPoolModelFromResponse(string resourceGroup, string serverName, Management.Sql.Models.ElasticPool pool)
         {
-            DatabaseEdition edition = DatabaseEdition.None;
-            Enum.TryParse<DatabaseEdition>(pool.Edition, out edition);
-
             AzureSqlElasticPoolModel model = new AzureSqlElasticPoolModel
             {
                 ResourceId = pool.Id,
                 ResourceGroupName = resourceGroup,
                 ServerName = serverName,
                 ElasticPoolName = pool.Name,
                 CreationDate = pool.CreationDate ?? DateTime.MinValue,
-                DatabaseDtuMax = pool.DatabaseDtuMax.Value,
-                DatabaseDtuMin = pool.DatabaseDtuMin.Value,
-                Dtu = pool.Dtu,
                 State = pool.State,
-                StorageMB = pool.StorageMB,
+                MaxSizeBytes = pool.MaxSizeBytes,
                 Tags =
                     TagsConversionHelper.CreateTagDictionary(TagsConversionHelper.CreateTagHashtable(pool.Tags), false),
                 Location = pool.Location,
-                Edition = edition,
-                ZoneRedundant = pool.ZoneRedundant
+                Edition = pool.Edition,
+                ZoneRedundant = pool.ZoneRedundant,
+                Sku = pool.Sku,
+                PerDatabaseSettings = pool.PerDatabaseSettings,
+                Capacity = pool.Sku.Capacity,
+                CurrentServiceLevelObjectiveName = pool.Sku.Name,
+                MinCapacity = pool.PerDatabaseSettings.MinCapacity,
+                MaxCapacity = pool.PerDatabaseSettings.MaxCapacity
             };
 
             return model;
         }
+
+        /// <summary>
+        /// Get elastic pool sku name based on tier
+        ///    Edition              | SkuName
+        ///    GeneralPurpose       | GP
+        ///    BusinessCritical     | BC
+        ///    Standard             | StandardPool
+        ///    Basic                | BasicPool
+        ///    Premium              | PremiumPool
+        /// </summary>
+        /// <param name="tier">Azure Sql elastic pool edition</param>
+        /// <returns>The sku name</returns>
+        public static string getPoolSkuName(string tier)
+        {
+            if (string.IsNullOrWhiteSpace(tier))
+                return null;
+
+            string value;
+            return _vcoreEditionDic.TryGetValue(tier, out value) ? value : string.Format("{0}Pool", tier);
+        }
+
+        /// <summary>
+        /// Check if the tier is Vcore tier.
+        /// </summary>
+        /// <param name="tier">Azure sql elastic pool edition</param>
+        /// <returns>True if it is Vcore tier</returns>
+        public static bool isVcorePool(string tier)
```

Comment:
PowerShell should not have this logic. It will be fragile in the future when new editions are added.